src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,258 @@
+import {AggregatorV3Interface} from "./external/AggregatorV3Interface.sol";
+import {ERC20} from "@openzeppelin/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/token/ERC20/utils/SafeERC20.sol";
+
+/// @title AaveV2CollectorContractConsolidation
+/// @author Llama
+/// @notice Contract to swap excess assets for USDC at a discount
+contract AaveV2CollectorContractConsolidation {
+    using SafeERC20 for ERC20;
+
+    uint256 public immutable USDC_DECIMALS;
+    uint256 public immutable ETH_USD_ORACLE_DECIMALS;
+
+    ERC20 public constant USDC = ERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant ETH_USD_FEED = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
+
+    address public constant ARAI = 0xc9BC48c72154ef3e5425641a3c747242112a46AF;
+    address public constant AAMPL = 0x1E6bb68Acec8fefBD87D192bE09bb274170a0548;
+    address public constant AFRAX = 0xd4937682df3C8aEF4FE912A96A74121C0829E664;
+    address public constant FRAX = 0x853d955aCEf822Db058eb8505911ED77F175b99e;
+    address public constant AUST = 0xc2e2152647F4C26028482Efaf64b2Aa28779EFC4;
+    address public constant SUSD = 0x57Ab1ec28D129707052df4dF418D58a2D46d5f51;
+    address public constant ASUSD = 0x6C5024Cd4F8A59110119C56f8933403A539555EB;
+    address public constant TUSD = 0x0000000000085d4780B73119b644AE5ecd22b376;
+    address public constant ATUSD = 0x101cc05f4A51C0319f570d5E146a8C625198e636;
+    address public constant AMANA = 0xa685a61171bb30d4072B338c80Cb7b2c865c873E;
+    address public constant MANA = 0x0F5D2fB29fb7d3CFeE444a200298f468908cC942;
+    address public constant ABUSD = 0xA361718326c15715591c299427c62086F69923D9;
+    address public constant BUSD = 0x4Fabb145d64652a948d72533023f6E7A623C7C53;
+    address public constant ZRX = 0xE41d2489571d322189246DaFA5ebDe1F4699F498;
+    address public constant AZRX = 0xDf7FF54aAcAcbFf42dfe29DD6144A69b629f8C9e;
+    address public constant AENS = 0x9a14e23A58edf4EFDcB360f68cd1b95ce2081a2F;
+    address public constant ADPI = 0x6F634c6135D2EBD550000ac92F494F9CB8183dAe;
+
+    event Swap(address indexed tokenIn, address indexed tokenOut, uint256 amountIn, uint256 amountOut);
+
+    /// Not enough token left to swap
+    error NotEnoughTokens();
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+    /// Token is not available for swap
+    error UnsupportedToken();
+
+    struct Asset {
+        uint256 quantity;
+        uint48 premium;
+        address oracle;
+        uint8 decimals;
+        uint8 oracleDecimals;
+        bool ethFeedOnly;
+    }
+
+    mapping(address => Asset) public assets;
+
+    constructor() {
+        assets[ARAI] = Asset(
+            ERC20(ARAI).balanceOf(AaveV2Ethereum.COLLECTOR),
+            100,
+            0x483d36F6a1d063d580c7a24F9A42B346f3a69fbb,
+            ERC20(ARAI).decimals(),
+            8,
```

Comment:
Since you're anyway doing this through the constructor and saving it. IMO, I think it makes sense to fetch the decimals directly from the Chainlink feed here instead of having it hardcoded for all these assets. It's easier to reason about that way and anyway looking at this code can be assured that it's getting the right decimals instead of having to cross-check it separately.